x/wasm/internal/keeper/test_common.go

```diff
@@ -412,3 +418,37 @@ func keyPubAddr() (crypto.PrivKey, crypto.PubKey, sdk.AccAddress) {
 	addr := sdk.AccAddress(pub.Address())
 	return key, pub, addr
 }
+
+var _ types.WasmerEngine = &AlwaysFailMockWasmer{}
+
+var MockDefaultError = errors.New("default mock test error")
+
+type AlwaysFailMockWasmer struct{}
+
+func (a AlwaysFailMockWasmer) Create(code cosmwasm.WasmCode) (cosmwasm.CodeID, error) {
+	return nil, MockDefaultError
+}
+
+func (a AlwaysFailMockWasmer) Instantiate(code cosmwasm.CodeID, env wasmTypes.Env, info wasmTypes.MessageInfo, initMsg []byte, store cosmwasm.KVStore, goapi cosmwasm.GoAPI, querier cosmwasm.Querier, gasMeter cosmwasm.GasMeter, gasLimit uint64) (*wasmTypes.InitResponse, uint64, error) {
+	return nil, 0, MockDefaultError
+}
+
+func (a AlwaysFailMockWasmer) Execute(code cosmwasm.CodeID, env wasmTypes.Env, info wasmTypes.MessageInfo, executeMsg []byte, store cosmwasm.KVStore, goapi cosmwasm.GoAPI, querier cosmwasm.Querier, gasMeter cosmwasm.GasMeter, gasLimit uint64) (*wasmTypes.HandleResponse, uint64, error) {
+	return nil, 0, MockDefaultError
+}
+
+func (a AlwaysFailMockWasmer) Query(code cosmwasm.CodeID, env wasmTypes.Env, queryMsg []byte, store cosmwasm.KVStore, goapi cosmwasm.GoAPI, querier cosmwasm.Querier, gasMeter cosmwasm.GasMeter, gasLimit uint64) ([]byte, uint64, error) {
+	return nil, 0, MockDefaultError
+}
+
+func (a AlwaysFailMockWasmer) Migrate(code cosmwasm.CodeID, env wasmTypes.Env, info wasmTypes.MessageInfo, migrateMsg []byte, store cosmwasm.KVStore, goapi cosmwasm.GoAPI, querier cosmwasm.Querier, gasMeter cosmwasm.GasMeter, gasLimit uint64) (*wasmTypes.MigrateResponse, uint64, error) {
+	return nil, 0, MockDefaultError
+}
+
+func (a AlwaysFailMockWasmer) GetCode(code cosmwasm.CodeID) (cosmwasm.WasmCode, error) {
+	return nil, MockDefaultError
+}
+
+func (a AlwaysFailMockWasmer) Cleanup() {
+	panic("implement me")
```

Comment:
Huh? That seems dangerous. Where is Cleanup called, just in test code?

Reply:
I would make this a no-op, I don't think we ever try to handle panics there.

Reply:
This would have been called from tests only